### DISCUSSION TITLE
Is the `clone` necessary in example for `clippy::needless_collect`?

### DISCUSSION THREAD

**Discussion by mickvangelderen** [rust-lang/rust-clippy#10784] [Q&A] [answered]
The example for [clippy::needless_collect](https://rust-lang.github.io/rust-clippy/master/#needless_collect) is written as follows:

```rust
let len = iterator.clone().collect::<Vec<_>>().len();
// should be
let len = iterator.count();
```

I have a feeling the the `.clone()` on the iterator should:
 - be present in both the before and after, or
 - be omitted from both.

Is this the case and should I open up a PR for this?

**Comment by Alexendoo** [comment, marked as answer]
Yeah it isn't needed, I'd go with the second option 👍

**Reply by mickvangelderen** [reply]
I suppose it was needed for the example to compile as it was written. I fixed this in the PR by redeclaring a new `iterator` and actually splitting the before and after block like in other examples.

**Comment by mickvangelderen** [comment]
Opened #10786
